### PR TITLE
Add more specific doc block type hints for arrays.

### PR DESCRIPTION
**Pull request by jwage**
While working on https://github.com/doctrine/skeleton-mapper/pull/28/files I found some more types that should be improved.

### CODE REVIEW

File: lib/Doctrine/Persistence/Mapping/Driver/FileDriver.php [hunk 1]
```diff
@@ -124,19 +124,17 @@ public function getAllClassNames() : array
...
+        /** @var array<string, ClassMetadata> $classCache */
         $classCache = $this->classCache;
 
-        /** @var string[] $keys */
+        /** @var array<int, string> $keys */
         $keys = array_keys($classCache);
 
-        /** @var string[] $merged */
-        $merged = array_unique(array_merge(
+        /** @var array<int, string> $merged */
```
Comment (Ocramius) [inline lib/Doctrine/Persistence/Mapping/Driver/FileDriver.php]: This variable no longer exists.

File: lib/Doctrine/Persistence/ObjectRepository.php [hunk 2]
```diff
@@ -34,10 +34,10 @@ public function findAll() : array;
      * an UnexpectedValueException if certain values of the sorting or limiting details are
      * not supported.
      *
-     * @param mixed[]       $criteria
-     * @param string[]|null $orderBy
+     * @param array<string, mixed> $criteria
+     * @param array<string, mixed> $orderBy
```
Comment (Ocramius) [inline lib/Doctrine/Persistence/ObjectRepository.php]: This was `string[]`: new format seems wrong

Comment (jwage) [reply lib/Doctrine/Persistence/ObjectRepository.php]: I changed it to `@param array<string, string> $orderBy`

File: lib/Doctrine/Persistence/Mapping/Driver/FileDriver.php (hunk 1 above)
Comment (jwage) [reply lib/Doctrine/Persistence/Mapping/Driver/FileDriver.php]: Removed.

**Pull request merged**